src/Microsoft.Health.Fhir.Core/Features/Persistence/BadRequestException.cs

```diff
@@ -18,5 +20,22 @@ public BadRequestException(string errorMessage)
                     OperationOutcomeConstants.IssueType.Invalid,
                     errorMessage));
         }
+
+        /// <summary>
+        /// Constructor creates BadRequestException from collection of error strings.
+        /// </summary>
+        /// <param name="errorMessages">Collection of error strings that are used to initialize OperationOutcomeIssue collection.
+        /// Must be non-null and non-empty</param>
+        public BadRequestException(ICollection<string> errorMessages)
+            : base((errorMessages.Count < 2) ? errorMessages.First() : "Multiple bad request errors.")
```

Comment:
I'm afraid I don't understand. Message itself is not for internal use, message will be returned to the client. I meant Exception itself is used internally so there's no need for a call param validation (at least not in the RELEASE code) since we control the caller code.

Reply:
If you really want, I can remove. But I really like it this way, makes it perfectly clear without worrying about operator precedence.

Reply:
> If you really want, I can remove the parentheses. But I really like it this way, makes it perfectly clear without worrying about operator precedence. It's just code style.

Totally fair! Happy to keep the extra parentheses if it improves readability.